core/DataTable/Filter/SafeDecodeLabel.php

```diff
@@ -10,13 +10,21 @@
 
 use Piwik\DataTable;
 use Piwik\DataTable\BaseFilter;
+use Piwik\Date;
 
 /**
  * Sanitizes DataTable labels as an extra precaution. Called internally by Piwik.
  *
  */
 class SafeDecodeLabel extends BaseFilter
 {
+    /**
+     * On this day we changed the default behavior of this filter. Before it called `urldecode()` on the label
+     * which would turn '+' signs, for instance, into spaces. Now we don't do this, but there may be very old
+     * archived data that has URL encoded labels. So for reports archived before this date, we decode the URL.
+     */
+    const DATE_OF_URLDECODE_CHANGE = '2018-09-25';
```

Comment:
this would imply that the user updated Matomo to that version on that day?

Reply:
no longer relevant